src/core/structured_config.py

```diff
@@ -235,3 +239,14 @@ def log_level_values(cls, value: str) -> str | None:
                 f"Value out of the accepted values. Could not properly parsed the roles configuration: {e}"
             )
         return value
+
+    @validator("roles", pre=True)
+    @classmethod
+    def roles_values(cls, value: str):
+        """Check roles values."""
+        roles = set(map(str.strip, value.split(",")))
```

Comment:
Sometimes, one-liners are good 🤝 I was thinking of returning `sorted(roles)` instead of `list(roles)`, just in case we wanted some guarantee on the roles order. Any thought?

Reply:
**nit:** I would expect the logic to follow `if role in roles_value: do something`, so sorting is not needed. Also, roles should never change within the context of a hook execution, so a `set` is the more "appropriate" type here I think.